pennylane/qchem/integrals.py

```diff
@@ -774,7 +774,7 @@ def _hermite_coulomb(t, u, v, n, p, dr):
     Returns:
         array[float]: value of the Hermite integral
     """
-    x, y, z = dr[0], dr[1], dr[2]
+    x, y, z = dr[0:3]
```

Comment:
Interesting, even though unpack each element anyways, the slicing is faster?

Reply:
Yes, seems to be faster.

Reply:
Out of curiosity, I did a small experiment:
```py
def test_A(x):
    x, y, z = x[0], x[1], x[2]
    return x + y + z

def test_B(x):
    x, y, z = x[0:3]
    return x + y + z
```

The representations are indeed slightly different:
`jax.make_jaxpr(test_A)(jax.numpy.array([1, 2, 3]))`
```
{ lambda ; a:i32[3]. let
    b:i32[1] = dynamic_slice[slice_sizes=(1,)] a 0
    c:i32[] = squeeze[dimensions=(0,)] b
    d:i32[1] = dynamic_slice[slice_sizes=(1,)] a 1
    e:i32[] = squeeze[dimensions=(0,)] d
    f:i32[1] = dynamic_slice[slice_sizes=(1,)] a 2
    g:i32[] = squeeze[dimensions=(0,)] f
    h:i32[] = add c e
    i:i32[] = add h g
  in (i,) }
```
vs
`jax.make_jaxpr(test_B)(jax.numpy.array([1, 2, 3]))`
```
{ lambda ; a:i32[3]. let
    b:i32[1] = slice[limit_indices=(1,) start_indices=(0,) strides=(1,)] a
    c:i32[] = squeeze[dimensions=(0,)] b
    d:i32[1] = slice[limit_indices=(2,) start_indices=(1,) strides=(1,)] a
    e:i32[] = squeeze[dimensions=(0,)] d
    f:i32[1] = slice[limit_indices=(3,) start_indices=(2,) strides=(1,)] a
    g:i32[] = squeeze[dimensions=(0,)] f
    h:i32[] = add c e
    i:i32[] = add h g
  in (i,) }
```

Timings are indeed faster for B:
```py
%%timeit -n 500
test_A(arr)
```
`1.15 ms ± 104 µs per loop (mean ± std. dev. of 7 runs, 500 loops each)`
```py
%%timeit -n 500
test_B(arr)
```
`78.2 µs ± 14.5 µs per loop (mean ± std. dev. of 7 runs, 500 loops each)`

Over an order of magnitude!

Reply:
Thanks David!

Reply:
🤯 This is so surprising to me